### PR TITLE
[feature-wip][multi-catalog] Support orc format file split for file scan node

### PR DESCRIPTION
**Pull request by SaintBacchus**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/exec/arrow/orc_reader.cpp [hunk 1]
```diff
@@ -52,6 +55,31 @@ Status ORCReaderWrap::init_reader(const TupleDescriptor* tuple_desc,
...
+        }
+        if (_range_start_offset + _range_size <= _offset) {
+            end_group = i;
+            break;
+        }
+    }
+    LOG(INFO) << "This reader read orc file from offset: " << _range_start_offset
+              << " with size: " << _range_size << ". Also mean that read from strip id from "
+              << _current_group << " to " << end_group;
+    _total_groups = end_group;
```
Comment (morningman) [inline be/src/exec/arrow/orc_reader.cpp:76]: if `_total_groups` is 0 here, than we don't need to call `_reader->Seek()`

File: be/src/exec/arrow/orc_reader.cpp [hunk 2]
```diff
@@ -52,6 +55,31 @@ Status ORCReaderWrap::init_reader(const TupleDescriptor* tuple_desc,
...
 
+    int64_t row_number = 0;
+    int end_group = _total_groups;
+    for (int i = 0; i < _total_groups; i++) {
+        int64_t _offset = _reader->GetRawORCReader()->getStripe(i)->getOffset();
+        int64_t row = _reader->GetRawORCReader()->getStripe(i)->getNumberOfRows();
+        if (_offset < _range_start_offset) {
+            row_number += row;
+        } else if (_offset == _range_start_offset) {
+            _current_group = i;
```
Comment (morningman) [inline be/src/exec/arrow/orc_reader.cpp:66]: `_current_group` may not be set here?
And it may always be 0?

**Review by morningman — COMMENTED** [review]
Compile error:
```
  /root/doris/be/src/exec/arrow/orc_reader.cpp:62:32: error: 'class arrow::adapters::orc::ORCFileReader' has no member named 'GetRawORCReader'
```

File: be/src/exec/arrow/orc_reader.cpp (hunk 2 above)
Comment (SaintBacchus) [reply be/src/exec/arrow/orc_reader.cpp:66]: In my test for orc file , `_range_start_offset` is one of the `_offset` list

File: be/src/exec/arrow/orc_reader.cpp (hunk 1 above)
Comment (SaintBacchus) [reply be/src/exec/arrow/orc_reader.cpp:76]: Yeah

**Review by morningman — APPROVED** [review]
LGTM

**Comment by carlvinhust2012** [comment]
> Compile error:
> 
> ```
>   /root/doris/be/src/exec/arrow/orc_reader.cpp:62:32: error: 'class arrow::adapters::orc::ORCFileReader' has no member named 'GetRawORCReader'
> ```

shall we re-compile the third-party to solve this error?

**Comment by SaintBacchus** [comment]
@carlvinhust2012 Yes, please use the latest docker image to compile it

**Pull request merged**